ldm/modules/encoders/modules.py

```diff
@@ -12,6 +12,15 @@
 )  # TODO: can we directly rely on lucidrains code and simply add this as a reuirement? --> test
 
 
+def get_default_device_type():
```

Comment:
we use this in so many places, could be good to factor it out into a file and import it into the scripts that need it.

Reply:
Turns out it is already in [simplet2i](https://github.com/lstein/stable-diffusion/blob/713b2a03dc2dbda12ea4908054d07efb0cdabaad/ldm/simplet2i.py#L537). txt2img.py and img2img don't use it though. We'd have to change those 2 to work with simplet2i. I'm not sure if the goal of this fork is to do that. It should probably be moved out of this file though.

In the original pull request to the CompVis repo someone commented that I should change this code to something like the following because torch.backends.mps isn't available on all platforms.

```
    def _get_device(self):
        if torch.cuda.is_available():
            return torch.device('cuda')
        elif hasattr(torch.backends, 'mps') and torch.backends.mps.is_available():
            return torch.device('mps')
        else:
            return torch.device('cpu')
```

Reply:
Sorry, I just entered a redundant reply. Agree with @magnusviri - this is code that should be refactored in the future. I'm OK with leaving it in so that txt2img and img2img work on Macs.